ledger/participant-integration-api/src/main/scala/platform/store/Conversions.scala

```diff
@@ -172,6 +173,28 @@ private[platform] object Conversions {
                 }.toArray
               ).toEither.left.map(t => TypeDoesNotMatch(t.getMessage))
 
+            case h2Array: H2JdbcArray =>
+              Try(
+                // H2 guarantees that getArray always returs an Object[] but for whatever
+                // reason the return type is still Object and we have to cast.
+                h2Array
+                  .getArray()
```

Comment:
Strange we would need this..H2 documentation claims mapping to Object[] directly (not to H2JdbcArray) :thinking:

Reply:
:shrug:

Reply:
oh so it turns out I was being stupid, while we do have an H2JdbcArray it inherits from `Array`. The issue was the varchar vs integer stuff not this. After switching to integer, I could just delete this.